runtime/src/accounts_db.rs

```diff
@@ -4592,9 +4639,7 @@ impl AccountsDb {
                 if count == 0 {
                     dead_slots.insert(*slot);
                 } else if self.caching_enabled
-                    && (self.page_align(store.alive_bytes() as u64) as f64
-                        / store.total_bytes() as f64)
-                        < SHRINK_RATIO
```

Comment:
Do we still want to keep this `SHRINK_RATIO` logic, or do we want to shrink every time we save at least a page of memory?

Reply:
Yea. I was debating this as well. The account store seems to stay tighter with this. It was around 28gb vs. 34gb for master. I was looking at the metrics for increased write IO. I think it is a bit higher, but it wasn't super noticeable in the graph. Ideally one could adjust it dynamically. https://github.com/solana-labs/solana/issues/17544